File: Lua/d5.lua
s=... -- input
-- Part 1
c=0 for x in s:gmatch"%S+"do c=c+(x:find((".-[aeiou]"):rep(3))and x:find"(.)%1"and not(x:find"ab"or x:find"cd"or x:find"pq"or x:find"xy")and 1 or 0)end print(c)
-- Part 2
c=0 for x in s:gmatch"%S+"do c=c+(x:find"(..).-%1"and x:find"(.).%1"and 1 or 0)end print(c)
-- Both parts
c=0 d=0 for x in s:gmatch"%S+"do c=c+(x:find((".-[aeiou]"):rep(3))and x:find"(.)%1"and not(x:find"ab"or x:find"cd"or x:find"pq"or x:find"xy")and 1 or 0)d=d+(x:find"(..).-%1"and x:find"(.).%1"and 1 or 0)end print(c,d)

File: Python/d5.py
s=... # input
# Part 1
import re;print(sum(all(map(re.search,('(.*[aeiou]){3}',r'(.)\1','^((?!ab|cd|pq|xy).)*$'),[u]*3))for u in s.split()))
# Part 2
import re;print(sum(all(map(re.search,(r'(..).*\1',r'(.).\1'),[u,u]))for u in s.split()))
# Both parts
import re;f=lambda*a:print(sum(all(map(re.search,a,[u]*len(a)))for u in s.split()));f('(.*[aeiou]){3}',r'(.)\1','^((?!ab|cd|pq|xy).)*$'),f(r'(..).*\1',r'(.).\1')
